script.js

```diff
@@ -1,4 +1,344 @@
+        const randomChoice = Object.values(Choices)[randomIndex];
+        runGame(randomChoice);
+      }
+      break;
+    default:
+      runGame(rawInput);
+      break;
+  }
+}
+
+function runGame(rawInput) {
+  const randomIndex = Math.floor(Math.random() * Object.keys(Choices).length);
+  const computerChoice = Object.values(Choices)[randomIndex];
+
+  let playerInput = null;
+  let winState = null;
+
+  Object.values(Choices).forEach((value) => {
+    if (rawInput.includes(value)) {
+      playerInput = value;
+    }
+  });
+
+  if (playerInput === null) {
+    displayResults("Please enter a valid input");
+  } else {
+    winState = getWinState(playerInput, computerChoice, programState.gameMode);
+    if (programState.gameMode != GameMode.Korean) {
+      calculateScore(playerInput, computerChoice, winState);
+    }
+  }
+}
+
+function getWinState(playerInput, computerChoice, gameMode) {
+  let winState = null;
+  // Check default win state
+  switch (playerInput) {
+    case Choices.Scissors:
+      {
+        if (computerChoice === Choices.Scissors) {
+          winState = GameEndState.Draw;
+        } else if (computerChoice === Choices.Stone) {
+          winState = GameEndState.Lose;
+        } else {
+          winState = GameEndState.Win;
+        }
+      }
+      break;
+    case Choices.Paper:
+      {
+        if (computerChoice === Choices.Paper) {
+          winState = GameEndState.Draw;
+        } else if (computerChoice === Choices.Scissors) {
+          winState = GameEndState.Lose;
+        } else {
+          winState = GameEndState.Win;
+        }
+      }
+      break;
+    case Choices.Stone:
+      {
+        if (computerChoice === Choices.Stone) {
+          winState = GameEndState.Draw;
+        } else if (computerChoice === Choices.Paper) {
+          winState = GameEndState.Lose;
+        } else {
+          winState = GameEndState.Win;
+        }
+      }
+      break;
+    default:
+      // NOTE: Error checking should be done before calling this function
+      break;
+  }
+
+  console.log("Checking state...");
+
+  // Special cases to change winState
+  switch (gameMode) {
+    case GameMode.Reversed:
+      {
+        if (winState !== GameEndState.Draw) {
+          winState = winState == GameEndState.Win ? GameEndState.Lose : GameEndState.Win;
+        }
+      }
+      break;
+    case GameMode.Korean:
+      {
+        let outputText = `You chose ${playerInput} and the computer chose ${computerChoice}.`;
+        outputText += "<br><br>";
+
+        if (programState.koreanGameState.isFinalRound === false) {
+          if (winState === GameEndState.Win) {
+            programState.koreanGameState.score = KoreanGameScore.PlayerWins;
+            programState.koreanGameState.isFinalRound = true;
+            outputText += "<b>You're winning so far!</b>";
+          } else if (winState === GameEndState.Lose) {
+            programState.koreanGameState.score = KoreanGameScore.ComputerWins;
```

Comment:
Tho not applicable for this game, I think a 
programState.koreanGameState.score += KoreanGameScore.ComputerWins; 
will fit better